plugin/trino-sqlserver/src/main/java/io/trino/plugin/sqlserver/SqlServerClient.java

```diff
@@ -441,6 +479,74 @@ public void setNull(PreparedStatement statement, int index)
         };
     }
 
+    public static ColumnMapping timestampColumnMapping(TimestampType type)
+    {
+        if (type.getPrecision() <= MAX_SHORT_PRECISION) {
+            // SQLServer JDBC driver does not support LocalDateTime so timestampColumnMapping() cannot be used here
+            return ColumnMapping.longMapping(
+                    type,
+                    shortTimestampReadFunction(),
+                    shortTimestampWriteFunction());
+        }
+        return ColumnMapping.objectMapping(
+                type,
+                longTimestampReadFunction(),
+                longTimestampWriteFunction());
+    }
+
+    private static LongReadFunction shortTimestampReadFunction()
+    {
+        return (resultSet, columnIndex) -> {
+            Timestamp timestamp = resultSet.getTimestamp(columnIndex, newUtcCalendar());
+            int nanoSeconds = timestamp.getNanos() % NANOSECONDS_PER_MILLISECOND;
+            verify(nanoSeconds % SQLSERVER_MAX_PRECISION_NANOSECONDS == 0, "Invalid value of nanoSeconds: %s", nanoSeconds);
```

Comment:
this should be more strict. If we have; TIMESTAMP(1) we should only allow single non-zero digit; for TIMESTAMP(2) two non zero digits and so on. 

I think that maybe we should add common helper verify methods:
`Timestamps.verifyShortTimestampValue(int precision, long value)`
`Timestamps.verifyLongTimestampValue(int precision, LongTimestamp value)`

Or alternatively `verifyValue` instance methods in `ShortTimesttmpType` and `LongTimestampType`

Reply:
@nineinchnick smartly noticed that if we put the verify methods in `-spi` we cannot use Guava as implementation blocks. 

@findepi is the `trino-plugin-toolkit` good place for those? E.g new class `Timestamps` in `/trino-plugin-toolkit/src/main/java/io/trino/plugin/base/type`. Or should we rather extend one of the classes in SPI? If the latter do you have a preference for one of the apporaches I proposed above?

Reply:
I added `verifyValue` in this class but I'm not sure how to properly test the negative cases, where the server would return higher precision than what's defined in Trino.

Reply:
I do not need to do so (and also do not know how :) ). Provided we test the `verifyValue` method itself.

Reply:
Added tests and I had to change the visibility to protected.

Reply:
@losipiuk i would need to understand why we need verify. But, since all these is currently JDBC-specific, it's ok to keep the verify helpers in base-jdbc, and we're good for now

Reply:
True - base-jdbc is good place for now too.

Reply:
I moved them to base-jdbc and split into short and long versions, with extra checks for precision range. Also moved and updated tests. All of this is in a separate commit, please review. Thanks!